pkg/inject/sidecar.go

```diff
@@ -199,6 +198,15 @@ func container(jaeger *v1.Jaeger, dep *appsv1.Deployment) corev1.Container {
 			agentTags = fmt.Sprintf("%s,%s=%s", agentTags,
 				"container.name", dep.Spec.Template.Spec.Containers[0].Name,
 			)
+		} else if agentIdx > -1 {
```

Comment:
I think the original condition isn't accurate anymore and needs to be fixed. We should rather get a list of containers with the agent filtered out, and then check the length. The case here is that we might have more containers in the pod besides the Jaeger Agent, and we wouldn't know which value to use as `container.name`.

That said, I think this whole conditional deserves a refactoring already. It should place all the tags as a `map[string]string` and join them once the map is complete. This way, we add the `container.name` to the map in only one place.